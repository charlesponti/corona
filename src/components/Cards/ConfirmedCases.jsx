import React from 'react'
import PropTypes from 'prop-types'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faVirus } from '@fortawesome/free-solid-svg-icons'

import Title from '../Title'
import CountUpTo from '../CountUpTo'
import Rate from '../Rate'
import DateText from '../DateText'
import cx from 'classnames'
import styles from './Cards.module.css'

export default function ConfirmedCases ({ confirmed, lastUpdate }) {
  const worldPopulation = 7594000000

  return (
    <Card className={cx(styles.card, styles.confirmed)}>
      <CardContent>
        <FontAwesomeIcon icon={faVirus} size="4x" style={{ marginBottom: '1rem' }}/>
        <Title title="Confirmed Cases" />
        <CountUpTo value={confirmed} />
        <Rate typeOfRate="Population" percentFor={confirmed} percentOf={worldPopulation} />
        <DateText date={lastUpdate} />
      </CardContent>
    </Card>
  )
}

ConfirmedCases.propTypes = {
  confirmed: PropTypes.number,
  lastUpdate: PropTypes.string
}
