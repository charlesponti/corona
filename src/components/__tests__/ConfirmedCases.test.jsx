import React from 'react'
import ConfirmedCases from '../Cards/ConfirmedCases'
import { render } from '@testing-library/react'

test('should render ConfirmedCases', async (done) => {
  const { container, findByText } = render(
    <ConfirmedCases confirmed={75940000} lastUpdate="2020-04-29T00:00:00.000z" />
  )
  expect(await findByText('75,940,000')).toBeTruthy()
  expect(await findByText('1.00 %')).toBeTruthy()
  expect(await findByText('Tue Apr 29 2020')).toBeTruthy()
  expect(container).toMatchSnapshot()
  done()
})
