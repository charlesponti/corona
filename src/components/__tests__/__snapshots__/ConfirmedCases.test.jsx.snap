// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConfirmedCases 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root card confirmed MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
        style="font-size: 3rem;"
      >
        <span
          aria-label="sick emoji"
          role="img"
        >
          ??
        </span>
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
      >
        Confirmed Cases
      </p>
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        <span>
          0
        </span>
      </h5>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        <b>
          Population
           Rate:
        </b>
         
        1.00 %
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        Tue Apr 28 2020
      </p>
    </div>
  </div>
</div>
`;
